openmeter/entitlement/entitlement.go

```diff
@@ -92,6 +97,100 @@ type CreateEntitlementInputs struct {
 	Config                  []byte                 `json:"config,omitempty"`
 	UsagePeriod             *UsagePeriod           `json:"usagePeriod,omitempty"`
 	PreserveOverageAtReset  *bool                  `json:"preserveOverageAtReset,omitempty"`
+
+	SubscriptionManaged bool `json:"subscriptionManaged,omitempty"`
+}
+
+func (c CreateEntitlementInputs) Equal(other CreateEntitlementInputs) bool {
+	if c.Namespace != other.Namespace {
+		return false
+	}
+
+	if !reflect.DeepEqual(c.FeatureID, other.FeatureID) {
+		return false
+	}
+
+	if !reflect.DeepEqual(c.FeatureKey, other.FeatureKey) {
```

Comment:
why do we need deep equal for a string?

Reply:
deepEqual is not ideal, its used because they're `*string` not `string`